Kernel/Storage/AHCIPort.cpp

```diff
@@ -19,9 +19,36 @@
 
 namespace Kernel {
 
-NonnullRefPtr<AHCIPort> AHCIPort::create(const AHCIPortHandler& handler, volatile AHCI::PortRegisters& registers, u32 port_index)
+static ALWAYS_INLINE bool is_interface_disabled(volatile AHCI::PortRegisters& registers)
 {
-    return adopt_ref(*new AHCIPort(handler, registers, port_index));
+    return (registers.ssts & 0xf) == 4;
+}
+
+KResultOr<NonnullRefPtr<AHCIPort>> AHCIPort::try_create(const AHCIPortHandler& handler, volatile AHCI::PortRegisters& registers, u32 port_index)
+{
+    if (is_interface_disabled(registers)) {
+        auto port = adopt_ref_if_nonnull(new AHCIPort(handler, registers, port_index));
+        if (!port)
+            return KResult(ENOMEM);
+        return port.release_nonnull();
```

Comment:
This is such a common pattern that this now exists :) (also there's the extra `(nothrow)`)
```suggestion
        return adopt_nonnull_ref_or_enomem(new (nothrow) AHCIPort(handler, registers, port_index));
```